variants/deploy_with_capistrano/template.rb

```diff
@@ -0,0 +1,179 @@
+puts "=" * 80
+puts "Starting variant: deploy_with_capistrano"
+puts "=" * 80
+
+append_to_file("Gemfile") do
+  <<~'EO_RUBY'
+
+    # Deployment
+    #
+    # We tell bundler not to require these gems by default because the code that
+    # requires them will explicitly require them.
+    #
+    gem "capistrano", require: false
+    gem "capistrano-bundler", require: false
+    gem "capistrano-rails", require: false
+    gem "capistrano-rbenv", require: false
+    gem "capistrano-rake", require: false
+
+    # Capistrano fails when run from Github Action CI unless these gems are
+    # present in the bundle. See https://github.com/net-ssh/net-ssh/issues/565
+    gem "ed25519", require: false
+    gem "bcrypt_pbkdf", require: false
+
+  EO_RUBY
+end
+
+run "bundle install"
+run "bundle exec cap install"
+
+old_generated_cap_config_snippet = <<~EO_RUBY
+  set :application, "my_app_name"
+  set :repo_url, "git@example.com:me/my_repo.git"
+EO_RUBY
+
+new_ackama_cap_config_snippet = <<~'EO_RUBY'
+
+  ##
+  # Uncomment and configure this block if your deployments go through a
+  # bastion/jump host.
+  #
+  # CAP_BASTION_USER = "deploy".freeze
+  # CAP_BASTION_HOST = "some-ip-or-hostname".freeze
+  # set :ssh_options, {
+  #   proxy: Net::SSH::Proxy::Command.new("ssh -o StrictHostKeyChecking=no #{CAP_BASTION_USER}@#{CAP_BASTION_HOST} -W %h:%p")
+  # }
+
+  set :application, "TODO_set_app_name"
+  set :repo_url, "TODO_set_git_repo_url"
+
+  set :bundle_config, {
+    deployment: true,
+    without: "development:test"
+  }
+
+  # Default value for :linked_files is []
+  append :linked_files, ".env"
+
+  # Default value for linked_dirs is []
+  append :linked_dirs, "log", "tmp/pids", "tmp/cache", "tmp/sockets", "public/system", "storage"
+
+  ##
+  # Roles
+  #
+  # We have a single :app role which contains all the servers we care about in
+  # this deployment.
+  #
+  # We run nginx, rails on the same servers so we do not need the distinction
+  # between the :web and :app roles that Capistrano encourages by default.
+  #
+  # Capistrano is not involved at all in Database management so we don't need a
+  # separate :db role either
+  #
+  set :assets_roles, [:app] # defaults to [:web]
+  set :migration_role, :app # defaults to :db
+
+  # Default value for default_env is {}
+  set :default_env, path: "$HOME/.rbenv/shims:$HOME/bin:/snap/bin:$PATH"
+
+  set :rbenv_path, "$HOME/.rbenv"
+  set :rbenv_ruby, File.read(".ruby-version").strip
+  set :rbenv_prefix, "$HOME/.rbenv/bin/rbenv exec"
+  set :rbenv_map_bins, %w[rake gem bundle ruby rails]
+
+  namespace :deploy do
+    desc "Restart application"
+    task :restart do
+      on roles(:app), in: :sequence, wait: 5 do
+        # TODO: replace these examples with the real commands that control your
+        # server
+        execute :sudo, "systemctl restart puma"
```

Comment:
I think it'd be neat if we added `execute :sudo, "systemctl restart sidekiq"` based on if the `sidekiq` variant was enabled, and then we could replace that `TODO` because we only want people to be using `puma` and I'd expect advanced setups that might include any other services to already be on top of this